### PR TITLE
Add test for unicode in regular strings

### PR DESCRIPTION
**Pull request by jchavarri**
Extracted from the discussions around #1141 and #1142.

Special emphasis on the final test that shows compatibility with OCaml when matching over "bytes as string", which justifies the escaping of regular string literals with `\x` and prevents using unicode characters like emojis in the generated JS code.

### CODE REVIEW

File: test/blackbox-tests/unicode-in-regular-strings.t [hunk 1]
```diff
@@ -0,0 +1,210 @@
+Showcase usage of escaped and literal unicode text in regular strings
+
+  $ . ./setup.sh
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.15)
```
Comment (anmonteiro) [inline test/blackbox-tests/unicode-in-regular-strings.t:6]: it's what's in the opam file

```suggestion
  > (lang dune 3.13)
```

File: test/blackbox-tests/unicode-in-regular-strings.t [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+  $ cat > x.ml <<EOF
+  > let bytes =
+  >   Bytes.init 4 (function
+  >     | 0 -> '\xf0'
+  >     | 1 -> '\x9f'
+  >     | 2 -> '\x90'
+  >     | 3 -> '\xab'
+  >     (* Should never happen since the length is 4 *)
+  >     | _ -> '\x00')
```
Comment (anmonteiro) [inline test/blackbox-tests/unicode-in-regular-strings.t:155]:
```suggestion
  >     | _ -> assert false)
```

**Review by anmonteiro — APPROVED** [review]
thanks for the comprehensive tests.

this is great documentation to point to when someone (like future us) invariably tries to fix this in a broken way
